### PR TITLE
Add Python Data Science

### PR DESCRIPTION
**Pull request by krzjoa**
**[https://github.com/krzjoa/awesome-python-data-science](https://github.com/krzjoa/awesome-python-data-science)**

**Comprehensive list of Python libraries for Data Science: Machine Learning, Data Analysis, Visualization, Probabilistic Modelling etc.**

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The suggested Awesome list complies with the below requirements.
  ✖   Missing main list heading                                     remark-lint:awesome/heading - **I used title in HTML tags**
  ✖  (several times) Titles should use ' as a quote                                remark-lint:link-title-style - **Don't know, why**
  ✖  Do not use upper-case characters in definition labels         remark-lint:definition-case - **Caused by R language label; used for the sake of convention**
  ✖   32:1    ToC item "Machine Learning" link "#ml" not found              remark-lint:awesome/toc - **it does work**
  ✖  (multipletimes)  List item description must end with proper punctuation        remark-lint:awesome/list-item - **because of .png badges used aftermany item descriptions**
  ✖  (several times)   List item description must start with valid casing            remark-lint:awesome/list-item - **I used for consisten pandas naming convention**

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- **If you use a license badge, it should be SVG, not PNG.**
	- To verify that you've read all the guidelines, please add comment with just the word `unicorn`.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

### CODE REVIEW

**Comment by krzjoa** [comment]
Unicorn :sunglasses:

**Comment by krzjoa** [comment]
Reviews: #1605 and #1600.

**Comment by sindresorhus** [comment]
Nice header!

---

The icons you use for some items should be high DPI, meaning set to half the original size. Right now, they look blurry on high DPI screens.

---

The repo name is `awesome-python-datascience`, but it should be `awesome-python-data-science`. Data science is not a single word.

---

```
README.md:25:1
  ✖   19:95   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   20:93   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   21:94   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   22:94   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   23:95   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   24:96   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   25:1    Do not use upper-case characters in definition labels         remark-lint:definition-case
  ✖   25:91   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   26:95   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   27:96   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   28:95   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   29:97   Titles should use ' as a quote                                remark-lint:link-title-style
  ✖   32:1    ToC item "Machine Learning" link "#ml" not found              remark-lint:awesome/toc
  ✖   74:60   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  101:79   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  102:75   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  106:47   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  113:56   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  114:71   List item description must start with valid casing            remark-lint:awesome/list-item
  ✖  115:43   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  129:52   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  142:46   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  144:45   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  145:47   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  163:63   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  185:57   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  214:41   List item description must start with valid casing            remark-lint:awesome/list-item
  ✖  242:75   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  266:65   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  284:65   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  288:43   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  289:47   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  295:65   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  313:45   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  319:45   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  326:54   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  328:32   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  353:50   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  355:73   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  360:48   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  364:67   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  366:61   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  383:92   Link to https://zpjn.wmi.amu.edu.pl is dead                   remark-lint:no-dead-urls
  ✖  385:45   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  416:59   List item description must start with valid casing            remark-lint:awesome/list-item
  ✖  431:73   List item description must start with valid casing            remark-lint:awesome/list-item
```

https://github.com/sindresorhus/awesome-lint

File: readme.md [hunk 1]
```diff
@@ -130,6 +130,7 @@
 	- [Asyncio](https://github.com/timofurrer/awesome-asyncio#readme) - Asynchronous I/O in Python 3.
 	- [Scientific Audio](https://github.com/faroit/awesome-python-scientific-audio#readme) - Scientific research in audio/music.
 	- [CircuitPython](https://github.com/adafruit/awesome-circuitpython#readme) - A version of Python for microcontrollers.
+	- [Data Science](https://github.com/krzjoa/awesome-python-datascience) - Data Analysis and Machine Learning
```
Comment (sindresorhus) [inline readme.md]: Please read the guidelines for the description.

**Comment by krzjoa** [comment]
I've added a dot at the end of the line.

**Comment by krzjoa** [comment]
* I renamed the repo to **awesome-python-data-science**
* All icons are now bigger .png files
* Awesome Lint is complaining about four following things:
  README.md:1:1
  ✖    1:1   Missing main list heading                           remark-lint:awesome/heading - **I used HTML header**
  ✖   21:1   ToC item "Machine Learning" link "#ml" not found    remark-lint:awesome/toc - **It exists and works**
  ✖  405:59  List item description must start with valid casing  remark-lint:awesome/list-item - **name 'pandas' starting with a uppercase in order to stick to convetion.**
  ✖  420:73  List item description must start with valid casing  remark-lint:awesome/list-item - **As above, but with scikit-learn in description**

File: readme.md [hunk 2]
```diff
@@ -130,6 +130,7 @@
 	- [Asyncio](https://github.com/timofurrer/awesome-asyncio#readme) - Asynchronous I/O in Python 3.
 	- [Scientific Audio](https://github.com/faroit/awesome-python-scientific-audio#readme) - Scientific research in audio/music.
 	- [CircuitPython](https://github.com/adafruit/awesome-circuitpython#readme) - A version of Python for microcontrollers.
+	- [Data Science](https://github.com/krzjoa/awesome-python-data-science) - Data Analysis and Machine Learning.
```
Comment (csabapalfi) [inline readme.md]: Please add #readme at the end of your link.

Comment (krzjoa) [reply readme.md]: Done 😎

**Comment by nglgzz** [comment]
There are a few other linting errors besides the ones you already mentioned:

```
  ✖  303:160  List item description must end with proper punctuation  
  ✖  402:60   List item description must end with proper punctuation  
  ✖  415:57   List item description must end with proper punctuation  
```

Also for this one:

```
 ✖   20:1    ToC item "Machine Learning" link "#ml" not found   
```

You should just change the link to "#machine-learning"

**Comment by krzjoa** [comment]
If it comes to the current linting results:
* _Missing main list heading_ - I have used a **HTML h1 tag**, and apparently it isn't traited as a header.
* _ToC item "Machine Learning" link "#ml" not found_ - ToC item exists in the list and the link **does work**.
* _303:160  List item description must end with proper punctuation_ - This one is a unsolved mistery :no_mouth:.
* I've resigned from **pandas** and **scikit-learn** at the beginning of the two descriptions to fullfil the severe (:wink:) rules.

**Comment by nglgzz** [comment]
> 303:160 List item description must end with proper punctuation - This one is an unsolved mistery :no_mouth:.

This one means that you missed the `.` at the end of the description, you have this on MXFusion, tensorpack, and PyTorchNet.

> ToC item "Machine Learning" link "#ml" not found - ToC item exists in the list and the link does work.

It might be just a coincidence that it's working, this is how a proper anchor should be made https://gist.github.com/asabaylus/3071099#gistcomment-1593627

**Comment by krzjoa** [comment]
Done! 😏

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1177089755969777665

**Comment by ManjuladeviM** [comment]
Great post.
[Data Science With R Online Certification ](ur=https://www.gangboard.com/big-data-training/data-science-with-r-trainingl)
